src/query/functions/src/aggregates/aggregate_distinct_state.rs

```diff
@@ -171,31 +179,58 @@ impl DistinctStateFunc<DataGroupValues> for AggregateDistinctState {
     }
 }
 
+impl AggregateDistinctStringState {
+    #[inline]
+    fn insert_and_materialize(&mut self, key: &KeysRef) {
+        let entity = self.set.insert_key(key, &mut self.inserted);
+        if self.inserted {
+            let data = unsafe { key.as_slice() };
+
+            let holder = self.holders.last_mut().unwrap();
+            if holder.may_resize(data.len()) {
```

Comment:
There will be memory fragmentation here, please add comment for refactor in future

Reply:
ok